common/k8stest/util_postgres_app.go

```diff
@@ -0,0 +1,314 @@
+package k8stest
+
+import (
+	"context"
+	"fmt"
+	"github.com/openebs/openebs-e2e/common/custom_resources"
+	"github.com/openebs/openebs-e2e/common/e2e_config"
+	"strings"
+	"time"
+
+	batchV1 "k8s.io/api/batch/v1"
+	coreV1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// PostgresApp represents a PostgreSQL application deployment configuration.
+type PostgresApp struct {
+	Namespace    string     // Kubernetes namespace where the PostgreSQL application is deployed.
+	Pod          coreV1.Pod // Kubernetes Pod running the PostgreSQL application.
+	VolUuid      string     // UUID of the provisioned volume for the PostgreSQL application.
+	ReleaseName  string     // Helm release name for the PostgreSQL application.
+	ReplicaCount int        // Number of replicas for the PostgreSQL StatefulSet.
+	ScName       string     // StorageClass name used for provisioning volumes.
+	Standalone   bool       // Indicates if the PostgreSQL deployment is standalone.
+	PvcName      string     // Name of the PersistentVolumeClaim.
+	PgBench      PgBenchApp // PgBench application configuration for benchmarking.
+}
+
+// PostgresInstallReady checks if the PostgreSQL application is installed and ready.
+func (psql *PostgresApp) PostgresInstallReady() error {
+	logf.Log.Info("checking postgres application to be installed")
+	ready := false
+	counter := 12
+	if psql.Standalone {
```

Comment:
What's behaviour when Postgres arch is not standalone ?

Reply:
We are not ready for that at the moment because we don't need it. All test cases are for a simple standalone version. We could add a second type, but I'll add it after the merge because we're running out of time and it's better and easier to test with the code after splitting the repo.

Reply:
Returned error with not implemented message otherwise